wear/src/main/java/io/homeassistant/companion/android/home/MainViewModel.kt

```diff
@@ -358,17 +367,46 @@ class MainViewModel @Inject constructor(
     fun addFavoriteEntity(entityId: String) {
         viewModelScope.launch {
             favoritesDao.addToEnd(entityId)
+            addCachedFavorite(entityId)
         }
     }
 
     fun removeFavoriteEntity(entityId: String) {
         viewModelScope.launch {
             favoritesDao.delete(entityId)
+            removeCachedFavorite(entityId)
+        }
+    }
+
+    fun getCachedEntity(entityId: String): FavoriteCaches? =
+        favoriteCaches.find { it.id == entityId }
+
+    private fun addCachedFavorite(entityId: String) {
+        viewModelScope.launch {
+            val entity = entities[entityId]
+            val attributes = entity?.attributes as Map<*, *>
+            val icon = attributes["icon"] as String?
+            favoriteCachesDao.add(FavoriteCaches(entityId, attributes["friendly_name"].toString(), icon))
```

Comment:
In case the friendly name isn't set nothing should be stored instead of the string `"null"`, which would be even uglier in my opinion. That will allow the app to show the entity ID instead like it does now.

(If you agree, make sure to also update the data class to reflect this.)

Reply:
Storing the entity ID _as_ the friendly name when there is no friendly name also works :)